src/gsi/m_berror_stats.f90

```diff
@@ -498,73 +689,107 @@ subroutine read_wgt(corz,corp,hwll,hwllp,vz,corsst,hsst,varq,qoption,varcw,cwopt
-            else
-               call setcorchem_(cvars3d(n),corz(:,:,n),ier)
-               if ( ier/=0 ) cycle ! if this happens, code will crash later
-            endif
-            call sethwlloz_(hwll(:,:,n),mype)
-            call setvscalesoz_(vz(:,:,n))
-         endif
-         if ( mype==0 ) write(6,*) myname_, ': WARNING, using general Berror template for ', cvars3d(n)
-      endif
-   enddo
-
-!  if so, overwrite cw-cov with q-cov
-   iq=-1;icw=-1
-   do n=1,size(cvars3d)
-      if(trim(cvars3d(n))=='q' ) iq =n
-      if(trim(cvars3d(n))=='cw') icw=n
-   enddo
-   if (cwcoveqqcov_) then
-      if(iq>0.and.icw>0) then
-        hwll(:,:,icw)=hwll(:,:,iq)
-        vz  (:,:,icw)=vz  (:,:,iq)
-      end if
-   end if
-   if (cwoption==1 .or. cwoption==3) then
-      if (iq>0.and.icw>0) then
-         do k=1,nsig
-            do i=1,nlat
-               corz(i,k,icw)=one
-            end do
-         end do
-         hwll(:,:,icw)=0.5_r_kind*hwll(:,:,iq)
-         vz  (:,:,icw)=0.5_r_kind*vz  (:,:,iq)
-      end if 
 
-      if (n_clouds_fwd>0 .and. icw<=0) then
-         do n=1,size(cvars3d)
-            do ic=1,n_clouds_fwd
-               if(trim(cvars3d(n))==trim(cloud_names_fwd(ic))) then
-                  ivar=n
-                  do k=1,nsig
-                     do i=1,nlat
-                        corz(i,k,ivar)=one
-                     end do
-                  end do
-                  hwll(:,:,ivar)=0.5_r_kind*hwll(:,:,iq)
-                  vz  (:,:,ivar)=0.5_r_kind*vz  (:,:,iq)
-                  exit
-               end if   
-            end do
-         end do
-      end if
-   endif
+  end subroutine bin_
 
-   ! need simliar general template for undefined 2d variables ...
+  subroutine nc_(myid)
 
-   deallocate(found3d,found2d)
+   integer,intent(in) :: myid
+   type(nc_berror_vars) bvars
+   real(r_single), pointer :: ptr1d(:)
+   real(r_single), pointer :: ptr2d(:,:)
+   integer :: nv 
+   call nc_berror_read (berror_stats,bvars,ier, myid=myid,root=0)
+   if (nlat/=bvars%nlat .or. nlon/=bvars%nlon .or.  nsig/=bvars%nsig ) then
+      call die(myname_," inconsistent dims in "//trim(berror_stats), 99)
+   endif
+   isig=bvars%nsig
+
+!  RTodling: the following is bad since it wires all naming conventions ... to be revised
+   do nv=1,size(cvars2d)
+      if (trim(cvars2d(nv))=='sst') then
+         n = getindex(cvars2d,'sst')
+         found2d(n)=.true.
+         call nc_berror_getpointer (cvars2d(nv),bvars,ptr2d,ier)
+         if(ier==0) corsst=ptr2d
+         call nc_berror_getpointer ('h'//cvars2d(nv),bvars,ptr2d,ier)
+         if(ier==0) hsst=ptr2d
+      endif
+      if (trim(cvars2d(nv))=='ps') then
+         n = getindex(cvars2d,'ps')
+         found2d(n)=.true.
+         call nc_berror_getpointer (cvars2d(nv),bvars,ptr1d,ier)
+         if(ier==0) corp(:,n)=ptr1d
+         call nc_berror_getpointer ('h'//cvars2d(nv),bvars,ptr1d,ier)
+         if(ier==0) hwllp(:,n)=ptr1d
+      endif
```

Comment:
is it worth adding a warning print / failure if a cvars2d is defined but no entry in the file is expected? or is it fine that this is hard coded for just these 2?

Reply:
Good point.  Only 2 of the 9 `call nc_berro_getpointer` have logic handling the `ier /= 0` case.  When `ier /= 0` is encountered for these 2 cases, program execution is terminated.  Error checking added for the 7 other  cases.  Done at 849b3b19